x/evm/keeper/attest_reference_block.go

```diff
@@ -0,0 +1,66 @@
+package keeper
+
+import (
+	"context"
+	"errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/palomachain/paloma/x/consensus/keeper/consensus"
+	consensustypes "github.com/palomachain/paloma/x/consensus/types"
+	"github.com/palomachain/paloma/x/evm/types"
+)
+
+func (k Keeper) attestReferenceBlock(ctx context.Context, q consensus.Queuer, msg consensustypes.QueuedSignedMessageI) (retErr error) {
+	if len(msg.GetEvidence()) == 0 {
+		return nil
+	}
+
+	_, chainReferenceID := q.ChainInfo()
+
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	logger := k.Logger(ctx).WithFields(
+		"chain-id", chainReferenceID,
+		"component", "attest-reference-block",
+		"msg-id", msg.GetId(),
+		"msg-nonce", msg.Nonce())
+	logger.Debug("attest-reference-block")
+
+	cacheCtx, writeCache := sdkCtx.CacheContext()
```

Comment:
I don't think this is needed here tbh. There's only 1 state change involved here, which we will only reach if all preconditions are fulfilled.

Reply:
I kept it as part of the wrapper as other handler might need it.